p4-16/spec/P4-16-spec.mdk

```diff
@@ -2777,6 +2777,22 @@ keep the least-significant W bits of the value). In particular, P4
 does not have arithmetic exceptions---the result of an arithmetic
 operation is defined for all possible inputs.
 
+P4 target architectures may also support saturating arithmetic. All saturating
+operations are limited to a fixed range between a minimum and maximum value.
+Saturating arithmetic has advantages, in particular when used as counters. The
+the result of a saturating counter max-ing out is much closer to the real
+result than a counter that overflows and wraps around. According to Wikipedia
+(https://en.wikipedia.org/wiki/Saturation_arithmetic) saturating arithmetic is
+as numerically close to the true answer as possible; for 8-bit binary signed
+arithmetic, when the correct answer is 130, it is considerably less surprising
+to get an answer of 127 from saturating arithmetic than to get an answer of
+−126 from modular arithmetic. Likewise, for 8-bit binary unsigned arithmetic,
+when the correct answer is 258, it is less surprising to get an answer of 255
+from saturating arithmetic than to get an answer of 2 from modular arithmetic.
+At this time, P4 defines saturating operations only for multiplication and
+addition. For an unsigned integer with bit-width of `W`, the minimum value is `0`
```

Comment:
Should "multiplication and addition" be replaced with "addition and subtraction"?  Those are the operators defined later.

Reply:
+1 on replacing multiplication with subtraction.

Reply:
fixed.